shared/controlflow/codeql/controlflow/Guards.qll

```diff
@@ -362,6 +362,12 @@ module Make<
     v.asConstantValue() = c.asConstantValue()
   }
 
+  private predicate trivialHasValue(Expr e, GuardValue v) {
+    constantHasValue(e, v)
+    or
+    e instanceof NonNullExpr and v.isNonNullValue()
```

Comment:
The reason why this is not pushed into `constantHasValue` is because that predicate is meant to only hold for values `v` that uniquely determine the value of `e`?